spec/models/recurring_transaction_rule_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe RecurringTransactionRule, type: :model do
+  describe "associations" do
+    it { is_expected.to belong_to(:wallet) }
+  end
+
+  describe "enums" do
+    it { is_expected.to define_enum_for(:interval).with_values(%i[weekly monthly quarterly yearly]) }
+    it { is_expected.to define_enum_for(:method).with_values(%i[fixed target]) }
+    it { is_expected.to define_enum_for(:trigger).with_values(%i[interval threshold]) }
+    it { is_expected.to define_enum_for(:status).with_values(%i[active terminated]) }
+  end
```

Comment:
I know it's very few it blocks here but as our test suite is getting slower (and bigger), I think we should be careful and create les examples when testing the same context

```suggestion
  it do
    expect(described_class).to belong_to(:wallet)
    expect(described_class).to define_enum_for(:interval).with_values(%i[weekly monthly quarterly yearly])
    expect(described_class).to define_enum_for(:method).with_values(%i[fixed target])
    expect(described_class).to define_enum_for(:trigger).with_values(%i[interval threshold])
    expect(described_class).to define_enum_for(:status).with_values(%i[active terminated])
  end
```